src/core/src/bootstrap/EnvLayer.py

```diff
@@ -430,26 +430,26 @@ def write_with_retry_using_temp_file(file_path, data, mode='w'):
                         raise Exception("Unable to write to {0} (retries exhausted). Error: {1}.".format(str(file_path), repr(error)))
 
         @staticmethod
-        def delete_files_from_dir(dir_name, file_identifier_list, raise_if_delete_failed=False):
-            """ Clears all files from given dir. NOTE: Uses file_identifier_list to determine the content to delete """
-            for file_identifier in file_identifier_list:
-                files_to_delete = glob.glob(str(dir_name) + "/" + str(file_identifier))
+        def delete_from_dir(dir_name, identifier_list, raise_if_delete_failed=False):
+            """ Clears all files/dirs from given dir. NOTE: Uses identifier_list to determine the content to delete """
+            for identifier in identifier_list:
+                items_to_delete = glob.glob(os.path.join(str(dir_name), str(identifier)))
 
-                for file_to_delete in files_to_delete:
+                for item_to_delete in items_to_delete:
                     try:
-                        os.remove(file_to_delete)
+                        if os.path.isdir(item_to_delete):
+                            shutil.rmtree(item_to_delete)
+                        else:
+                            os.remove(item_to_delete)
                     except Exception as error:
-                        error_message = "Unable to delete files from directory [Dir={0}][File={1}][Error={2}][RaiseIfDeleteFailed={3}].".format(
-                            str(dir_name),
-                            str(file_to_delete),
-                            repr(error),
-                            str(raise_if_delete_failed))
+                        error_message = "Unable to delete item from directory [Dir={0}][Item={1}][Error={2}][RaiseIfDeleteFailed={3}].".format(
+                            str(dir_name), str(item_to_delete), repr(error), str(raise_if_delete_failed))
 
                         if raise_if_delete_failed:
                             raise Exception(error_message)
                         else:
                             print(error_message)
-                            return None
+                            continue
```

Comment:
@rane-rajasi - notes: not using continue made this whole thing fail on the first failure without processing additional entries.

Reply:
IIRC, it was intentionally designed in that way. But I'm okay with this